### PR TITLE
Fix save button blinking

### LINKED ISSUE #618: [Bug]: Blinking save btn when active list is only modified visually
### Release Type

Compiled Windows x86 64-bit

### Version

v1.0.10 edge 51

### Operating System

Win 11

### Relevant logs

_No response_

### Describe the bug

Currently, the save btn flashes when filters are changed for the active list (searching mods, changing shown mod types) even when there is no real change to the actual modlist. Additionally, the flashing does not go away until a save action is done, even if the filter actions are undone.

Note: This bug doesn't seem to be 100% consistent. Sometimes it occurs, othertimes the expected behavior occurs.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Duplicate Issue Check

- [X] I've checked for duplicate issues

### Wiki/FAQ Check

- [X] I've checked the Wiki for a solution

### PR DESCRIPTION
**Pull request by Patryk-Malinowski**
Closes #618

When saving the modlist, it didnt save the mod uuids to a list that it used for comparison later hence triggering the animation. The save list seemed to be updated on refresh only (or if opening up rimsort, because it calls the ```_do_refresh``` method)

This means if you open up rimsort and start using the filters, it works as expected. But once you make changes and 'save' is when the issues can be reproduced.

Video replicating the issue.

https://github.com/user-attachments/assets/7cad045f-073a-4b50-8659-615f905e3578

No harm checking if the save button animation can be triggered differently from what Is shown in the video to make sure this fix still addresses it...

### CODE REVIEW

**Comment by Patryk-Malinowski** [comment]
Additionally save the modlists to their restore states/variables as previously it was only done on startup.

**Review by LionelColaso — APPROVED** [review]
Works nice catch

File: app/views/main_content_panel.py [hunk 1]
```diff
@@ -1761,6 +1761,8 @@ def _do_save(self) -> None:
                         active_mods.append(package_id + "_steam")
                         continue  # Append `_steam` suffix if Steam mod, continue to next mod
                 active_mods.append(package_id)
+        active_mods_uuids, inactive_mods_uuids, _, _ = metadata.get_mods_from_list(mod_list=[mod for mod in active_mods])
```
Comment (hk21702) [inline app/views/main_content_panel.py:1764]: Have not been able to test on my own, but is it necessary to make a copy of the active_mods list?

Comment (Patryk-Malinowski) [reply app/views/main_content_panel.py]: No it is not. Great catch.

**Pull request merged**